update_cache.py

```diff
@@ -64,10 +65,31 @@ def main():
         sys.exit(1)
 
     # let's grab sha and version info as Tuxsuite has the most up-to-date info
-    # we only need the first entry
-    first_entry = builds[next(iter(builds))]
-    git_sha = first_entry["git_sha"]
-    clang_version = first_entry["tuxmake_metadata"]["compiler"]["version_full"]
+    builds_that_are_missing_metadata = []
+    git_sha = None
+    clang_version = None
+    for entry, build in builds.items():
+        try:
+            git_sha = build["git_sha"]
+            clang_version = build["tuxmake_metadata"]["compiler"][
+                "version_full"]
+        except KeyError:
+            builds_that_are_missing_metadata.append(entry)
+
+    if len(builds_that_are_missing_metadata) == len(builds):
+        raise RuntimeError(
+            f"Could not find a suitable git sha or compiler version in any build\n"
+            f"Here's the build.json:\n{raw}")
+
+    if len(builds_that_are_missing_metadata) > 0:
+        print(
+            "Warning: Some of the builds in builds.json are malformed and missing "
+            "some metadata.\n"
+            f"Here's a list: {builds_that_are_missing_metadata}\n"
+            f"Here's the build.json in question:\n{raw}")
+        sys.exit(0)
```

Comment:
Shouldn't this `sys.exit(0)` be removed so that we go and update the cache?

Reply:
Gotcha, checkout [5069c26](https://github.com/ClangBuiltLinux/continuous-integration2/pull/683/commits/5069c260af95a8872a3e181ceb823428c2d5b705)